src/string.ts

```diff
@@ -105,26 +105,24 @@ export default class StringSchema<
     });
   }
 
-  matches(regex: RegExp, options: MatchOptions | MatchOptions['message']) {
+  matches(regex: RegExp, options: MatchOptions | MatchOptions['message'] = locale.matches) {
```

Comment:
this is intentionally not being defaulted up here because it can mess with the overload check below, e.g. messages can be objects. I'd just mark this as optional instead

Reply:
Ok. Reverted and just marked it optional.